app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts

```diff
@@ -172,40 +208,55 @@ export class TestRunner {
   }
 
   public async runTests() {
-    const summary: TestSummary = {
-      passed: 0,
-      failed: 0,
-      failedTests: [] as Array<string>,
-      startTime: Date.now(),
-      endTime: 0,
-    };
-
-    const previousNestingLevel = this._nestingLevel;
+    console.log('\n');
 
     for (const testSuite of this._testSuites) {
-      await this.runTestSuite(testSuite, summary);
+      this._currentTestSuite = testSuite;
+      await testSuite.buildSuite();
+      this._currentTestSuite = null;
+    }
+    for (const testSuite of this._testSuites) {
+      await this.runTestSuite(testSuite);
     }
-    this._nestingLevel = previousNestingLevel;
 
     this._testSuites = [];
     console.log('End of tests run 🏁');
-    summary.endTime = Date.now();
-    this.printSummary(summary);
+    this._summary.endTime = Date.now();
+    this.printSummary();
   }
 
-  private async runTestSuite(testSuite: TestSuite, summary: TestSummary) {
-    this._currentTestSuite = testSuite;
-    this._nestingLevel = testSuite.nestingLevel;
+  private async runTestSuite(testSuite: TestSuite) {
+    if (testSuite.skip) {
+      this._summary.skipped += testSuite.testCases.length;
+      return;
+    }
 
-    console.log(`${indentNestingLevel(this._nestingLevel)} ${testSuite.name}`);
+    if (this._includesOnly) {
+      let skipTestSuite = !testSuite.only;
+      for (const testCase of testSuite.testCases) {
+        if (testCase.only) {
+          skipTestSuite = false;
+        }
+      }
+      if (skipTestSuite) {
+        this._summary.skipped += testSuite.testCases.length;
+        return;
+      }
+    }
+
+    this._currentTestSuite = testSuite;
+    console.log(`${indentNestingLevel(testSuite.nestingLevel)} ${testSuite.name}`);
 
-    testSuite.buildSuite();
     if (testSuite.beforeAll) {
       await testSuite.beforeAll();
     }
 
     for (const testCase of testSuite.testCases) {
-      await this.runTestCase(testSuite, testCase, summary);
+      if ((!this._includesOnly || testSuite.only || testCase.only) && !testCase.skip) {
```

Comment:
It seems a bit too complicated. We currently have four different flags to determine if a test case should be executed or not: `this._includesOnly`, `testSuite.only`, `testCase.only`, and `testCase.skip`. What we can do is simplify this by using just one flag, for instance `testCase.skip`. Instead of removing test cases using `this._testSuites.splice`, each test case can be explicitly assigned the `skip` property. This way, we only need to check if a test should be skipped in one specific location. Currently, the `runTestSuite` function has two additional early returns to handle skipping.

In summary, let's simplify things. We will utilize a single flag to decide whether a test case should be executed or not, and we will check this flag in just one place.

Reply:
The `splice` function is not needed to remove tests from test suits, but to insert them in order. Unfortunately, since we build testSuites before running them, it is needed to keep them in order. 
When we append each test suite at the end the order of the nested suites is wrong.

Reply:
I've added an additional loop to iterate over all the decorators and correctly assign skip depending on other flags.

Reply:
I can't get rid of the splice function inside `describe`, since the order of tests would be wrong. Yet I can propose the following changes to simplify the code, which I would like to implement:
* Move logic to find index in describe into separate util file
* Create folder testRunner, and rename testRunner into testRunner/index.ts, so we could store all testRunnerHelpers in one place (asserts, Matchers, stringFormatUtils and so on)
* Have all the decorators grouped in additional object
* Move function printSummary to stringUtils etc.
* Move notificationRegistry and the related functions to a separate file.

I would like to do it in separate PR to reduce the number of conflicts with my implementation of decorators `failing` i `warning`